engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -200,7 +205,12 @@ namespace dmGameSystem
         }
         world->m_ComponentIndex = params.m_ComponentIndex;
         world->m_3D = physics_context->m_3D;
-        world->m_Components.SetCapacity(32);
+        uint32_t comp_count = params.m_MaxComponentIntances;
+        if (comp_count == 0xFFFFFFFF)
+        {
+            comp_count = 32;
+        }
+        world->m_Components.SetCapacity(comp_count);
```

Comment:
We don't have any max_count for physics component in game.project and dynamicly change capacity of this array.
So, in this case If I know exact count of component I use it insted of default initial valus which is 32. But if this component may be spawned by factory (0xFFFFFFFF) I use 32 as a staring value.